lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -2001,7 +2001,7 @@ def _clients_list(self) -> list:
         """
         clients = []  # type: list
         for relation in self._charm.model.relations.get(self._relation_name, []):
-            clients = clients + json.loads(relation.data[relation.app]["endpoints"])
+            clients = clients + json.loads(relation.data[relation.app].get("endpoints", "[]"))
```

Comment:
Is this change still required ? It seems to be out of sync with current main branch. Please see [this line](https://github.com/canonical/loki-k8s-operator/blob/main/lib/charms/loki_k8s/v0/loki_push_api.py#L2011) on main branch.

Reply:
The line you linked if bugged out big times. Passing a `""` to JSON parsing and trying to treat it as a list to concatenate to another? :D